### ISSUE TITLE
Support chained ogg files

### ISSUE THREAD

**Issue by est31** [RustAudio/lewton#18]
Apparently its legal to chain multiple logical streams inside an ogg/vorbis file. The vorbis spec explicitly allows it:

```
The Ogg/Vorbis I specification currently dictates that Ogg/Vorbis streams use Ogg transport streams in degenerate, unmultiplexed form only. That is:

    A meta-headerless Ogg file encapsulates the Vorbis I packets
    The Ogg stream may be chained, i.e., contain multiple, contigous logical streams (links).
    The Ogg stream must be unmultiplexed (only one stream, a Vorbis audio stream, per link) 
```

See also similar bugs filed for [flac](https://trac.videolan.org/vlc/ticket/14972), in [firefox](https://bugzilla.mozilla.org/show_bug.cgi?id=455165), and [opus](https://trac.videolan.org/vlc/ticket/18401).

We should at least support playback here.

**Comment by est31** [comment]
The use case is streaming apparently, when you want to update metadata while the stream is running.

**Comment by lu-zero** [comment]
In general the demuxer should issue what I call a new stream event and react to it by either allocating a new decoder if it is a new stream or resetting the current decoder pinned to the current stream.

If you have some test sample we could try to adapt your ogg demuxer to fit this model and make sure it works well enough.

**Comment by est31** [comment]
I don't have any test samples but I think you should be able to generate some by concatenating two distinct ogg/vorbis files (it is important that they are distinct so that the stream serials don't match).

The `ogg` crate and the low level API of lewton that rust-av uses are more or less ready for it. The `ogg` crate just spits out packets and the only state that the low level API holds must be destroyed one way or another. What does need a change is the glue code between the two. lewton provides its own glue code via the `inside_ogg` module. This glue code is not used by the av-vorbis crate however, more or less by construction (av-vorbis intends to be generic from the container, supporting e.g. matroska, while `inside_ogg` only concerns ogg). The purpose of the `inside_ogg` module is to allow "normal" users to parse ogg/vorbis files without much fuss which is 99% of what people want to do. So given that even if it is fixed in lewton, it won't automatically be fixed for rust-av, I think we should open a separate bug in the av-vorbis crate and track the issue there.

**Comment by est31** [comment]
There are test vectors in https://people.xiph.org/~xiphmont/test-vectors/vorbis/ (files `chain-test1.ogg` to `chain-test3.ogg`) . Commit e051c7efba1e023e7cdcf6e64396fb7a8ff66947 introduced initial support.

**Comment by est31** [comment]
The first and second chained file examples are now fixed with commits dbfaae5ad07edf322a7e41a43b2ecc2f7504f8ed and e2b38e849bd51e21a7c30baff7d5f2d0bd9d0653.
